scifin/classifier/classifier.py

```diff
@@ -89,32 +91,42 @@ def dtw_distance(ts1, ts2, window=None):
     """
     
     # Checks
-    try:
-        assert(ts1.type=='TimeSeries' and ts2.type=='TimeSeries')
-    except TypeError:
-        raise TypeError("Series have to be of type TimeSeries.")
-        
+    if not isinstance(ts1, ts.TimeSeries) and not isinstance(ts2, ts.TimeSeries):
+        raise AssertionError("Series have to be of type TimeSeries.")
+    if not isinstance(mode, str) and not mode in ['abs', 'square']:
+        raise AssertionError("mode must be a string, either 'abs' or 'square'.")
+
     # Initializations
+    # Window size
     N1 = len(ts1.data.index.tolist())
     N2 = len(ts2.data.index.tolist())
-    if window is not None:
-        assert(isinstance(window, int))
-        w = window
-    else:
-        w = N2
-    
+    if window is None:
+        window = N2
+    w = max(window, abs(N2-N1))
+
+    # Prepare dtw matrix
     dtw = np.full(shape=(N1+1,N2+1), fill_value=np.inf)
     dtw[0,0] = 0
+    for i in range(1, N1+1, 1):
+        for j in range(max(1,i-w), min(N2,i+w)+1, 1):
+            dtw[i,j] = 0
 
     # Loop
-    for i in range(0, N1, 1):
-        # for j in range(0, N2, 1):
-        for j in range(max(0,int(i-w)), min(N2,int(i+w)), 1):
-            square = (ts1.data.values[i] - ts2.data.values[j])**2
-            dtw[i+1,j+1] = square + min(dtw[i,j+1], dtw[i+1,j], dtw[i,j])
+    for i in range(1, N1+1, 1):
+        for j in range(max(1,i-w), min(N2,i+w)+1, 1):
+            if mode=='abs':
+                cost = abs(ts1.data.values[i-1] - ts2.data.values[j-1])
+            elif mode=='square':
+                cost = (ts1.data.values[i-1] - ts2.data.values[j-1])**2
+            dtw[i,j] = cost + min(dtw[i-1,j], dtw[i,j-1], dtw[i-1,j-1])
+    if verbose:
+        print(dtw)
             
     # Return distance
-    return np.sqrt(dtw[N1, N2])
+    if mode=='abs':
+        return dtw[N1, N2]
+    elif mode=='square':
+        return np.sqrt(dtw[N1, N2])
```

Comment:
so 'square' mean square root? I would prefer a different name than 'square' here

Reply:
No, it means the "cost" is a square distance. In fact, all the dtw matrix elements are square distances (and hence we need to take a sqrt at the end to still have a distance). I think from this point of view the name is appropriate, but I'm open to a better one.